### DISCUSSION TITLE
Get cid in http endpoint

### DISCUSSION THREAD

**Discussion by sinnbeck** [postalserver/postal#2587] [Help with using Postal]
Is there a way to get the cid of attachments? Currently it seems impossible to match attachments to their cid counterparts

![image](https://github.com/postalserver/postal/assets/13980140/8095d9fc-8d4e-41a5-a7fb-30639ad18089)

**Comment by willpower232** [comment]
As you say, they don't seem to be emitted currently. Can you find them in the database?

**Reply by sinnbeck** [reply]
I found a way of doing it. I send the raw mail and then use a php package to parse it instead :)

https://mail-mime-parser.org/

```php
$mailParser = new MailMimeParser();
$message = $mailParser->parse(base64_decode($message), false);
$cid = $message->getAttachmentPart(0)->getContentId(); //can be done in a foreach with getAllAttachmentParts()
